develop/dev-guide-sample-application-python-mysqlclient.md

```diff
@@ -1,292 +1,284 @@
+For more information, refer to [Query Data](/develop/dev-guide-get-data-from-single-table.md).
 
-If you are using a TiDB Serverless cluster, change the `get_connection` function in `mysqlclient_example.py`:
+### Update data
 
 ```python
-def get_connection(autocommit: bool = True) -> MySQLdb.Connection:
-    return MySQLdb.connect(
-        host="127.0.0.1",
-        port=4000,
-        user="root",
-        password="",
-        database="test",
-        autocommit=autocommit
-    )
+with get_mysqlclient_connection(autocommit=True) as conn:
+    with conn.cursor() as cur:
+        player_id, amount, price="1", 10, 500
+        cursor.execute(
+            "UPDATE players SET goods = goods + %s, coins = coins + %s WHERE id = %s",
+            (-amount, price, player_id),
+        )
 ```
 
-Suppose that the password you set is `123456`, and the connection parameters you get from the cluster details page are the following:
-
-- Endpoint: `xxx.tidbcloud.com`
-- Port: `4000`
-- User: `2aEp24QWEDLqRFs.root`
+For more information, refer to [Update Data](/develop/dev-guide-update-data.md).
 
-In this case, you can modify the `get_connection` as follows:
+### Delete data
 
 ```python
-def get_connection(autocommit: bool = True) -> MySQLdb.Connection:
-    return MySQLdb.connect(
-        host="xxx.tidbcloud.com",
-        port=4000,
-        user="2aEp24QWEDLqRFs.root",
-        password="123456",
-        database="test",
-        autocommit=autocommit,
-        ssl_mode="VERIFY_IDENTITY",
-        ssl={
-            "ca": "<ca_path>"
-        }
-    )
+with get_mysqlclient_connection(autocommit=True) as conn:
+    with conn.cursor() as cur:
+        player_id = "1"
+        cursor.execute("DELETE FROM players WHERE id = %s", (player_id,))
 ```
 
-### Step 3.3 Run the code
+For more information, refer to [Delete Data](/develop/dev-guide-delete-data.md).
 
-Before running the code, use the following command to install dependencies:
+## Useful notes
 
-```bash
-pip3 install -r requirement.txt
-```
+### Using driver or ORM framework?
 
-If you need to run the script multiple times, follow the [Table initialization](#step-31-initialize-table) section to initialize the table again before each run.
+The Python driver provides low-level access to the database, but it requires the developers to:
 
-```bash
-python3 mysqlclient_example.py
-```
+- Manually establish and release database connections.
+- Manually manage database transactions.
+- Manually map data rows (represented as tuples in `mysqlclient`) to data objects.
+
+Unless you need to write complex SQL statements, it is recommended to use [ORM](https://en.wikipedia.org/w/index.php?title=Object-relational_mapping) framework (like [SQLAlchemy](/develop/dev-guide-sample-application-python-sqlalchemy.md), [Peewee](/develop/dev-guide-sample-application-python-peewee.md) and Django ORM) for development. It can help you:
```

Comment:
```suggestion
Unless you need to write complex SQL statements, it is recommended to use [ORM](https://en.wikipedia.org/w/index.php?title=Object-relational_mapping) framework for development, such as [SQLAlchemy](/develop/dev-guide-sample-application-python-sqlalchemy.md), [Peewee](/develop/dev-guide-sample-application-python-peewee.md), and Django. It can help you:
```